gcp/src/main/java/org/apache/iceberg/gcp/auth/GoogleAuthManager.java

```diff
@@ -0,0 +1,160 @@
+
+  private GoogleCredentials credentials;
+  private boolean initialized = false;
+
+  public GoogleAuthManager(String managerName) {
+    this.name = managerName;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  private void initialize(Map<String, String> properties) {
+    if (initialized) {
+      return;
+    }
+
+    String credentialsPath = properties.get(GCPProperties.GCP_CREDENTIALS_PATH_PROPERTY);
+    String scopesString =
+        properties.getOrDefault(GCPProperties.GCP_SCOPES_PROPERTY, DEFAULT_SCOPES);
+    List<String> scopes = ImmutableList.copyOf(scopesString.split(","));
+
+    try {
+      if (credentialsPath != null && !credentialsPath.isEmpty()) {
+        LOG.info("Using Google credentials from path: {}", credentialsPath);
+        try (FileInputStream credentialsStream = new FileInputStream(credentialsPath)) {
+          this.credentials = GoogleCredentials.fromStream(credentialsStream).createScoped(scopes);
+        }
+      } else {
+        LOG.info("Using Application Default Credentials with scopes: {}", scopesString);
+        this.credentials = GoogleCredentials.getApplicationDefault().createScoped(scopes);
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to load Google credentials", e);
+    }
+    this.initialized = true;
+  }
+
+  /**
+   * Initializes and returns a short-lived session, typically for fetching configuration. This
+   * implementation reuses the long-lived catalog session logic.
+   */
+  @Override
+  public AuthSession initSession(RESTClient initClient, Map<String, String> properties) {
+    return catalogSession(initClient, properties);
+  }
+
+  /**
+   * Returns a long-lived session tied to the catalog's lifecycle. This session uses Google
+   * Application Default Credentials or a specified service account.
+   *
+   * @param sharedClient The long-lived RESTClient (not used by this implementation for credential
+   *     fetching).
+   * @param properties Configuration properties for the auth manager.
+   * @return A {@link GoogleAuthSession}.
+   * @throws UncheckedIOException if credential loading fails.
+   */
+  @Override
+  public AuthSession catalogSession(RESTClient sharedClient, Map<String, String> properties) {
+    initialize(properties);
+    Preconditions.checkState(
+        credentials != null, "GoogleAuthManager not initialized or failed to load credentials");
+    return new GoogleAuthSession(credentials);
+  }
+
+  /** Returns a session for a specific context. Defaults to the catalog session. */
+  @Override
+  public AuthSession contextualSession(SessionCatalog.SessionContext context, AuthSession parent) {
+    // For GCP, tokens are typically not context-specific in this manner.
+    // Re-using the parent (which should be a GoogleAuthSession) is appropriate.
+    // Or, if properties for a specific context were available, a new GoogleAuthSession could be
+    // derived.
+    if (parent instanceof GoogleAuthSession) {
```

Comment:
Imo this will always be true, so this whole method could be simplified to:

```java
public AuthSession contextualSession(SessionCatalog.SessionContext context, AuthSession parent) {
  return parent;
}
```